exercises/jigsaw-puzzle/canonical-data.json

```diff
@@ -0,0 +1,134 @@
+{
+  "exercise": "jigsaw-puzzle",
+  "comments": [
+    "You need to take only a few bits of information and provide the rest",
+    "Selecting a data format for the input and the output is important"
+  ],
+  "cases": [
+    {
+      "uuid": "123e44cd-1388-11f0-96bc-145afc5ea279",
+      "description": "1000 pieces puzzle with 1.6 aspect ratio",
+      "property": "jigsawData",
+      "input": {
+        "pieces": 1000,
+        "aspectRatio": 1.6
+      },
+      "expected": {
+        "pieces": 1000,
+        "border": 126,
+        "inside": 874,
+        "rows": 25,
+        "columns": 40,
+        "aspectRatio": 1.6,
+        "format": "portrait"
+      }
+    },
+    {
+      "uuid": "1da97b91-1388-11f0-a658-145afc5ea279",
+      "description": "square puzzle with 32 rows",
+      "property": "jigsawData",
+      "input": {
+        "rows": 32,
+        "format": "square"
+      },
+      "expected": {
+        "pieces": 1024,
+        "border": 124,
+        "inside": 900,
+        "rows": 32,
+        "columns": 32,
+        "aspectRatio": 1.0,
+        "format": "square"
+      }
+    },
+    {
+      "uuid": "25e21469-1388-11f0-881d-145afc5ea279",
+      "description": "300 pieces portrait puzzle with 70 border pieces",
+      "property": "jigsawData",
+      "input": {
+        "pieces": 300,
+        "border": 70,
+        "format": "portrait"
+      },
+      "expected": {
+        "pieces": 300,
+        "border": 70,
+        "inside": 230,
+        "rows": 25,
+        "columns": 12,
+        "aspectRatio": 0.48,
+        "format": "portrait"
+      }
+    },
+    {
+      "uuid": "2d910864-1388-11f0-9a22-145afc5ea279",
+      "description": "puzzle globe with 81 pieces",
+      "property": "jigsawData",
+      "input": {
+        "border": 0,
+        "inside": 82
```

Comment:
How is there no border? How are there more inside pieces than total pieces? 
Does this add a bunch of complexity/ambiguity to the problem? Is that worth adding? If yes, it might be worth describing what this case is all about.

Reply:
> How is there no border?

It is a globe.

> How are there more inside pieces than total pieces?

A typo.